### PR TITLE
Adds accountExists to XpringClient

### PR DESCRIPTION
**Pull request by amiecorso**
## High Level Overview of Change
Adds new function accountExists to XpringClient, as well as unit and integration tests.

### Context of Change
accountExists was written as a utility function the xpring-wallet-system, but should belong to the XpringClient directly. 

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)

## Before / After
accountExists now implemented

## Test Plan
CI

### CODE REVIEW

File: src/main/java/io/xpring/xrpl/XpringClient.java [hunk 1]
```diff
@@ -73,4 +73,13 @@ public String send(
     ) throws XpringException {
         return decoratedClient.send(amount, destinationAddress, sourceWallet);
     }
+    /**
```
Comment (keefertaylor) [inline src/main/java/io/xpring/xrpl/XpringClient.java]: nit: one vertical space above.

File: src/test/java/io/xpring/xrpl/DefaultXpringClientTest.java [hunk 2]
```diff
@@ -310,6 +312,64 @@ public void submitTransactionWithFailedSubmit() throws IOException, XpringExcept
...
+        DefaultXpringClient client = getClient(
+                accountInfoResult,
+                GRPCResult.ok(makeTransactionStatus(true, TRANSACTION_STATUS_SUCCESS)),
+                GRPCResult.ok(makeGetFeeResponse(MINIMUM_FEE, LAST_LEDGER_SEQUENCE)),
+                GRPCResult.ok(makeSubmitTransactionResponse(TRANSACTION_HASH))
+        );
+
+        // WHEN the existence of the account is checked
+        boolean exists = client.accountExists(XRPL_ADDRESS);
+        // THEN false is returned.
```
Comment (keefertaylor) [inline src/test/java/io/xpring/xrpl/DefaultXpringClientTest.java]: nit: +1 vertical space here

File: src/test/java/io/xpring/xrpl/legacy/LegacyDefaultXpringClientTest.java [hunk 3]
```diff
@@ -328,6 +332,84 @@ public void transactionStatusWithNodeError() throws IOException {
...
+                accountInfoResult,
+                GRPCResult.ok(makeFee(DROPS_OF_XRP_FOR_FEE)),
+                GRPCResult.ok(makeSubmitSignedTransactionResponse(TRANSACTION_BLOB)),
+                GRPCResult.ok(makeLedgerSequence()),
+                GRPCResult.ok(makeTransactionStatus(true, TRANSACTION_STATUS_SUCCESS))
+        );
+
+        // WHEN the existence of the account is checked
+        boolean exists = client.accountExists(XRPL_ADDRESS);
+        // THEN false is returned.
```
Comment (keefertaylor) [inline src/test/java/io/xpring/xrpl/legacy/LegacyDefaultXpringClientTest.java]: ditto

File: src/test/java/io/xpring/xrpl/legacy/LegacyDefaultXpringClientTest.java [hunk 4]
```diff
@@ -328,6 +332,84 @@ public void transactionStatusWithNodeError() throws IOException {
...
+                accountInfoResult,
+                GRPCResult.ok(makeFee(DROPS_OF_XRP_FOR_FEE)),
+                GRPCResult.ok(makeSubmitSignedTransactionResponse(TRANSACTION_BLOB)),
+                GRPCResult.ok(makeLedgerSequence()),
+                GRPCResult.ok(makeTransactionStatus(true, TRANSACTION_STATUS_SUCCESS))
+        );
+
+        // WHEN the existence of the account is checked
+        boolean exists = client.accountExists(XRPL_ADDRESS);
+        // THEN false is returned. (Legacy protobufs return UNKNOWN error code in case of now found account info)
```
Comment (keefertaylor) [inline src/test/java/io/xpring/xrpl/legacy/LegacyDefaultXpringClientTest.java]: ditto

Comment (amiecorso) [reply src/test/java/io/xpring/xrpl/legacy/LegacyDefaultXpringClientTest.java]: sweet! took care of the nits

**Pull request merged**